server/controllers/notification.js

```diff
@@ -0,0 +1,52 @@
+const asyncHandler = require("express-async-handler");
+const Notification = require("../models/Notification");
+
+// @route POST /notification/create
+// @desc create a notification
+// @access Private
+exports.createNotification = asyncHandler(async (req, res, next) => {
+    const userId = req.user.id;
+    const { type, title, description, context } = req.body;
+    const notification = await Notification.create({ userId, type, title, description, context });
+    res.status(201).json({ message: 'Notification created', notification });
```

Comment:
Wrap in a `try` block and return a different message on failure.

Reply:
added try-catch block in new changes.